chasm/tree.go

```diff
@@ -901,7 +922,7 @@ func (n *Node) IsDirty() bool {
 }
 
 func (n *Node) isValueSynced() bool {
-	if n.value != nil && !n.valueSynced {
+	if !n.valueSynced {
```

Comment:
With your definition of `valuedSynced` (which is bidirectional, n.value <-> n.serializedNode), we can't use this method in `IsDirty()`. Since a tree with only serializedNode but no decoded valued is not really "Dirty" from DB pov.